lib/response.rb

```diff
@@ -0,0 +1,23 @@
+class Response
+  attr_reader :verb, :path, :protocol, :host, :port, :origin, :accept
+
+  def initialize
+    @verb     = ""
+    @path     = ""
+    @protocol = ""
+    @host     = ""
+    @port     = ""
+    @origin   = ""
+    @accept   = ""
+  end
+
+  def process_lines(request_lines)
+    @verb     = request_lines[1][0]
+    @path     = request_lines[1][1]
+    @protocol = request_lines[1][2]
+    @host     = request_lines.last[1].split(':')[0]
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.